test/tests/functional/pbs_qsub_opts_args.py

```diff
@@ -135,28 +135,41 @@ def test_qsub_with_option_a(self):
         self.server.manager(MGR_CMD_SET, NODE,
                             {'resources_available.ncpus': 2},
                             self.mom.shortname)
-        # submit a job with future time and should start whenever time hits
+        ret_msg = 'qsub: Failed to save job/resv, '\
+            'refer server logs for details'
         present_tm = int(time.time())
+        # submit a job with future time and should start whenever time hits
         future_tm = time.strftime(
             "%Y%m%d%H%M", time.localtime(
                 present_tm + 60))
-        qsub_cmd = [self.qsub_cmd, '-a', future_tm, '--', '/bin/sleep', '100']
-        rv = self.du.run_cmd(self.server.hostname, cmd=qsub_cmd)
-        if rv['rc'] != 0:
-            self.fail("ERROR in submitting a job with future time: %s" % rv)
-        self.server.expect(JOB, {'job_state': 'W'}, id=str(rv['out'][0]))
+        j1 = Job(TEST_USER, {ATTR_a: future_tm})
+        try:
+            jid_1 = self.server.submit(j1)
+        except PbsSubmitError as e:
+            if e.rc == 57:
```

Comment:
I believe the fact that e.rc == 57 is a bug in PTL.  I wouldn't trust that it will always remain so.  @hirenvadalia can confirm this.

Reply:
@bhroam I don't think it is a bug in PTL because error code 57 is returned by qsub (as per the chat with @Bhagat-Rajput). So PTL is simply returning error code returned by qsub. However, first we need to find why qsub is returning 57 and what is the meaning of 57? @Bhagat-Rajput can you please find out why qsub is returning 57 and what is the meaning of that?

Reply:
Had a chat with @suresh-thelkar we came to know that the no. 57 is the result of 15161%256. As @bhroam said we also believe that these things should be handle at framework level.

Reply:
The framework can't handle this.  A mod operator is one way.  The framework can't take 57 and figure out it was 15161.  How about adding a comment in the test explaining what 57 means?

Reply:
As per my understanding, if PTL is using PTL_API mode then we should get 15161 error code only(The value should not be masked). If PTL is using PTL_CLI then we should do the mod operator as @bhroam suggested.

Reply:
As of now @suresh-thelkar PTL_API mode is not working due to some bugs so we cannot check that mode. I'll fix the PTL using one of the solution as @bhroam suggested.